src/AppInstallerCLICore/ExecutionReporter.h

```diff
@@ -165,9 +165,30 @@ namespace AppInstaller::CLI::Execution
             m_progressSink = sink;
         }
 
+        void SetLevelEnabled(Level reporterLevel, bool setEnabled = true) {
+            const bool isEnabled = levelEnabled(reporterLevel);

```

Comment:
```
void SetLevelEnabled(Level reporterLevel, bool setEnabled = true)
{
    if (setEnabled)
    {
        WI_SetFlag(m_enabledLevels, reportLevel)
    }
    else
    {
        WI_ClearFlag(m_enabledLevels, reportLevel)
    }
}

Reply:
This makes a lot of sense, I see what you're getting at and how the flag (bitmask) works. What I don't quite get is how to make the Level enum as a flag

Reply:
[Code](https://github.com/microsoft/winget-cli/blob/c5fbdd9fb9d53336bad182b82f255b2e3f691b10/src/AppInstallerCommonCore/Public/AppInstallerProgress.h#L41) to enable flags

Reply:
You're the best; Should have updates for this PR in a bit

Reply:
You can define `All = Verbose | Info | Warning | Error` in the enum definition to make life easier as well

Reply:
Okay, this is getting to be an area where I clearly have no idea how anything works anymore. I've specified the bitmasks, and dropped in the `DEFINE_ENUM_FLAG_OPERATORS`, which results in `linkage specification is not allowed`

Then, in `SetLevelEnabled` I get the error `expression must have a constant value` pointing to `reporterLevel` which cannot be used as a constant value

Reply:
Sorry for the confusion, please use WI_SetAllFlags and WI_ClearAllFlags for variables. WI_SetFlag and WI_ClearFlag require the second parameter to be compile time constant.

Reply:
> Sorry for the confusion, please use WI_SetAllFlags and WI_ClearAllFlags for variables. WI_SetFlag and WI_ClearFlag require the second parameter to be compile time constant.

That seemed to resolve that error, but there still seems to be something weird going on with the `DEFINE_ENUM_FLAG_OPERATORS`. If I put it before the Reporter struct, then I get that the `Reporter::Level` must be initialized first, but if I put it after the reporter struct then the functions inside the struct try to use an incorrect flag. If I put it inside the struct, I get the linkage error

![image](https://github.com/microsoft/winget-cli/assets/12611259/315225e2-55f4-4f11-8da0-b16c0af07b33)

Reply:
I just moved the enum outside the struct, since that seemed cleanest